pkg/core/blockchain.go

```diff
@@ -301,11 +301,16 @@ func (bc *Blockchain) AddBlock(block *block.Block) error {
 	if expectedHeight != block.Index {
 		return fmt.Errorf("expected block %d, but passed block %d", expectedHeight, block.Index)
 	}
+
+	headerLen := bc.headerListLen()
+	if int(block.Index) == headerLen {
+		err := bc.addHeaders(bc.config.VerifyBlocks, block.Header())
+		if err != nil {
+			return err
+		}
+	}
 	if bc.config.VerifyBlocks {
 		err := block.Verify()
```

Comment:
I think we should split this part of verification too, because merkle tree hash validity can (and should) be checked with the header alone (and transactions presence too).

Reply:
Oopsie, not the case, trimmed block is not a header and header is not a trimmed block. It's just not possible with header alone.